RR.App.Tests/Controllers/HRIS/ClientProjectsControllerUnitTests.cs

```diff
@@ -0,0 +1,198 @@
+            var createdAtActionResult = Assert.IsType<CreatedAtActionResult>(result);
+            Assert.Equal(nameof(_controller.GetClientProject), createdAtActionResult.ActionName);
+            Assert.Equal(createdClientProjectDto.Id, createdAtActionResult.RouteValues["id"]);
+            Assert.Equal(createdClientProjectDto, createdAtActionResult.Value);
+        }
+
+        [Fact]
+        public async Task UpdateClientProject_WithNonMatchingId_ReturnsBadRequest()
+        {
+            var clientProjectsDto = new ClientProjectsDto { Id = 2 };
+            var result = await _controller.UpdateClientProject(1, clientProjectsDto);
+            Assert.IsType<BadRequestObjectResult>(result);
+        }
+
+        [Fact]
+        public async Task SaveClientProject_WithInvalidModel_ReturnsBadRequest()
+        {
+            var clientProjectsDto = new ClientProjectsDto { };
+            _controller.ModelState.AddModelError("error", "sample error");
+
+            var result = await _controller.SaveClientProject(clientProjectsDto);
+            var badRequestResult = Assert.IsType<BadRequestObjectResult>(result);
+            Assert.Equal("Unable to create the project.", badRequestResult.Value);
+        }
+
+        [Fact]
+        public async Task GetClientProjectNotFoundResult()
+        {
+            _mockClientProjectService.Setup(service => service.GetClientProject(99))
+                .ReturnsAsync((ClientProjectsDto)null);
+
+            var result = await _controller.GetClientProject(99);
+            Assert.IsType<NotFoundResult>(result.Result);
+        }
+
+        [Fact]
+        public async Task UpdateClientProjectReturnsNotFoundResultOnException()
+        {
+            var clientProjectToUpdate = new ClientProjectsDto
+            { 
+                Id = 99, 
+                EmployeeId = 99,
+                ClientName = "string",
+                ProjectName = "string",
+                EndDate = DateTime.Now,
+                StartDate = DateTime.Now,
+                ProjectURL = "string",
+            };
+
+            _mockClientProjectService.Setup(service => service.UpdateClientProject(clientProjectToUpdate))
+                .ThrowsAsync(new Exception("An error occurred while updating the project."));
+
+            var result = await _controller.UpdateClientProject(99, clientProjectToUpdate);
+            var notFoundResult = Assert.IsType<ObjectResult>(result);
+            Assert.Equal("An error occurred while updating the project.", notFoundResult.Value);
+        }
+
+        [Fact]
+        public async Task ClientProjectReturnsNotFoundResultOnException()
+        {
+            _mockClientProjectService.Setup(service => service.DeleteClientProject(999))
+                .ThrowsAsync(new KeyNotFoundException("Project not found"));
+
+            var result = await _controller.DeleteClientProject(999);
+            Assert.IsType<NotFoundObjectResult>(result);
```

Comment:
keynotfound thrown but checking for notfound object result

Reply:
No